src/lender/token/memberlist.sol

```diff
@@ -0,0 +1,39 @@
+// Copyright (C) 2020 Centrifuge
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.5.15 <0.6.0;
+
+import "tinlake-math/math.sol";
+import "tinlake-auth/auth.sol";
+
+contract Memberlist is Math, Auth {
+
+    uint constant minimumDelay = 7 days; 
+
+    // -- Members--
+    mapping (address => uint) public members;
+    function updateMember(address usr, uint validUntil) public auth { 
+        require((safeAdd(now, minimumDelay)) < validUntil);
+        members[usr] = validUntil;
+     }
+
+    constructor() public {
+        wards[msg.sender] = 1;
+    }
+   
+    function member(address usr) public {
+        require((members[usr] >= now), "not-allowed-to-hold-token");
```

Comment:
Why not returning `true` or `false` here?

It is hard to build logic on top if it would fail here. 
```
if member {
  do x 
} else
{
do y
}
```